### DISCUSSION TITLE
Has anyone tried to use QTab, QTabs and QTabPanel from Quasar?

### DISCUSSION THREAD

**Discussion by scepeda78** [elimintz/justpy#231] [Q&A]
Has anyone tried to use QTab, QTabs and QTabPanel from Quasar?, I'm having issues to put the content inside of the tabs... Someone has an example of that?

**Comment by elimintz** [comment]
QTabPanels is not supported https://justpy.io/quasar_tutorial/not_working/

QTabs and QTab work.

**Comment by scepeda78** [comment]
Hi thanks for the quick answer!!

I was not able to find examples with QTab and QTabs. I know QTabPanels is
not working but QTabPanel works!!

The problem is that I cannot find examples of putting content inside of
tabs...

**Reply by matthoendorf** [reply]
Here is an example:

```
import justpy as jp

def qtab_click(self, msg):
    """function runs when user clicks a QTab"""
    # do something here
    print(f'tab name={self.value} was clicked')

def main():
    wp = jp.QuasarPage()

    # create tabs
    tabs = jp.QTabs(a=wp, classes='text-white shadow-2 q-mb-md', style="background-color: #00305e;", 
                    align='justify', narrow_indicator=True, dense=True)
    tab1 = jp.QTab(a=tabs, name='tab_1', label='Tab 1')
    tab2 = jp.QTab(a=tabs, name='tab_2', label='Tab 2')
    tab3 = jp.QTab(a=tabs, name='tab_3', label='Tab 3')
    tabs.on('input', qtab_click)

    return wp

jp.justpy(main)

```

**Reply by Lefort-Antoine** [reply]
Hy,
Thanks for this example.
Do you know how to create vertical tabs ? 
Regards

**Reply by matthoendorf** [reply]
@Lefort-Antoine I don't know if this will work, but try adding `vertical=True` as an attribute, i.e.:

```python
    tabs = jp.QTabs(a=wp, classes='text-white shadow-2 q-mb-md', style="background-color: #00305e;", 
                    align='justify', narrow_indicator=True, dense=True, vertical=True)
```

I base that on the example here: https://quasar.dev/vue-components/tabs#vertical

If that doesn't work, the next thing I would try is parsing the HTML:

```python
vertical_tab_example_html = """
<q-tabs
  v-model="tab"
  vertical
  class="text-teal"
>
  <q-tab name="mails" icon="mail" label="Mails" />
  <q-tab name="alarms" icon="alarm" label="Alarms" />
  <q-tab name="movies" icon="movie" label="Movies" />
</q-tabs>
"""
tabs = jp.parse_html(vertical_tab_example_html, a=wp)
```

Again, I haven't tested this code so it might require some fiddling.

**Reply by Lefort-Antoine** [reply]
Thanks for help
